src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -158,12 +158,11 @@ doReadRow d k = forModuleNameFix $ \mnFix ->
     lookupInPendingData (Utf8 rowkey) p = do
         let deltaKey = SQLiteDeltaKey tableNameBS rowkey
         ddata <- fmap _deltaData <$> MaybeT (return $ HashMap.lookup deltaKey (_pendingWrites p))
-        if null ddata
+        case ddata of
             -- should be impossible, but we'll check this case
-            then mzero
-            -- we merge with (++) which should produce txids most-recent-first
-            -- -- we care about the most recent update to this rowkey
-            else MaybeT $ return $! decodeStrict' $ DL.head ddata
+            [] -> mzero
```

Comment:
Can we get some kind of error message here?

Reply:
Now that you mention it, I think this should be a NonEmpty so that this case is impossible anyway.